### PR TITLE
Input required prop fixes.

### PR DESCRIPTION
**Pull request by morethanfire**
This is fore the church_management's PR [3329](https://github.com/saddlebackdev/church-management/pull/3329). 

Changes: 
- Added `componentDidUpdate` to set a new `showRequiredIndicator` state and to rerender the component with that value.
- Fixed majority of the Eslint errors.

Nothing in this should cause a regression except for maybe the `componentDidUpdate` function. Just to be aware of.

### CODE REVIEW

File: src/atoms/input.jsx [hunk 1]
```diff
@@ -324,88 +252,232 @@ class Input extends React.PureComponent {
...
+                newType = 'tel';
+                break;
+            case 'text':
+            default:
+                newType = 'text';
+                break;
+        }
+
+        return newType;
+    }
```
Comment (groberts314) [inline src/atoms/input.jsx:295]: Seems to me that, in theory, we shouldn't really need this, unless consumers are passing invalid `type` props.

Comment (morethanfire) [reply src/atoms/input.jsx:295]: Should remove? Is it a blocker?

**Comment by groberts314** [comment]
Testing Microsoft Teams integration.  This is a comment on a PR in `react-cm-ui` repo.

File: src/atoms/input.jsx (hunk 1 above)
Comment (groberts314) [reply src/atoms/input.jsx:295]: Nah ... definitely a NAB.  Up to you whether to remove now (and possibly incur small, random regressions if consumers are using non-standard values).  Maybe safer to keep it.  I imagine at some point there was a good reason for it.  Those React CM UI consumers can be sloppy coders sometimes! 😁

**Comment by morethanfire** [comment]
Oh, no! MS Teams!

**Pull request merged**